src/test/java/com/google/api/codegen/testdata/py/python_no_path_templates.baseline

```diff
@@ -928,16 +893,23 @@ class NoTemplatesAPIServiceClient(object):
             google.api_core.exceptions.RetryError: If the request failed due
                     to a retryable error and retry attempts failed.
         """
-        if 'increment' not in self._inner_api_calls:
-            self._inner_api_calls['increment'] = google.api_core.gapic_v1.method.wrap_method(
-                self._no_templates_api_service_stub.Increment,
+        # Wrap the transport method to add retry and timeout logic.
+        #
+        # Note: We use the hash of the original transport method
+        # in order to automatically invalidate the lookup if the
+        # transport method changes in-place (this can happen because
+        # the stubs must be regenerated if interceptors are added.)
+        cache_key = hash(self.transport.increment)
```

Comment:
This is *super* gross. This is breaking fundamental assumptions about the language and working around them in unfortunate ways. Transport should be immutable.

Reply:
The problem this is working around is that the stubs are created with the transport's channel, which can then change. I agree with you that this is not ideal, however.

One alternate idea would be to put the wrapper on interceptors on the primary client (I believe it is there already) and then simply make that write _a new transport object_ to the client. That would solve all of the issues on the transport and make it be frozen.

It does that at the expense of making the client's transport mutable, which feels pretty bad (but perhaps less bad).

Reply:
Would we be able to get around this simply by forcing interceptor binding to happen prior to the first API call a client makes?